sidekick_core/lib/sidekick_core.dart

```diff
@@ -134,15 +135,80 @@ class SidekickCommandRunner<T> extends CommandRunner<T> {
     exitCode = 0;
 
     final unmount = mount();
+
+    late final bool isSidekickUpdateCommand;
     try {
-      final result = await super.run(args);
+      final parsedArgs = parse(args);
+      final command = parsedArgs.command;
+      isSidekickUpdateCommand = command?.name == 'sidekick' &&
+          command is ArgResults &&
+          command.command?.name == 'update';
```

Comment:
Could we check for the `command` type? That would make it even more resilient and allows users to place the `SidekickCommand` anywhere in the command tree